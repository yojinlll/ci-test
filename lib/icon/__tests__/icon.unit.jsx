import renderer from 'react-test-renderer'
import React from 'react'
import Icon from '../Icon'
import {mount} from 'enzyme'

describe('Icon', ()=>{
  it('render svg ', ()=>{
    const json = renderer.create(<Icon name='github'/>).toJSON()
    expect(json).toMatchSnapshot()
  })
  it('onClick ', ()=>{
    const fn = jest.fn()
    const component = mount(<Icon name='github' onClick={fn}/>)
    console.log(component.find('svg'))
    component.find('svg').simulate('click')
    expect(fn).toBeCalled()
  })
})