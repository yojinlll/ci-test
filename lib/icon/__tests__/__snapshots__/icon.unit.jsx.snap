// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon render svg  1`] = `
<svg
  className="miro-icon"
>
  <use
    xlinkHref="#github"
  />
</svg>
`;
